os_stub/spdm_device_secret_lib_sample/lib.c

```diff
@@ -2221,65 +2222,75 @@ bool libspdm_write_certificate_to_nvm(
 #endif
     uint8_t slot_id, const void * cert_chain,
     size_t cert_chain_size,
-    uint32_t base_hash_algo, uint32_t base_asym_algo)
+    uint32_t base_hash_algo, uint32_t base_asym_algo
+#if LIBSPDM_SET_CERT_PARAMS
+    , bool *need_reset, bool *is_busy
+#endif /* LIBSPDM_SET_CERT_PARAMS */
+    )
 {
-#if defined(_WIN32) || (defined(__clang__) && (defined (LIBSPDM_CPU_AARCH64) || \
-    defined(LIBSPDM_CPU_ARM)))
-    FILE *fp_out;
-#else
-    int64_t fp_out;
-#endif
+    if (g_set_cert_is_busy) {
+        *is_busy = true;
 
-    char file_name[] = "slot_id_0_cert_chain.der";
-    /*change the file name, for example: slot_id_1_cert_chain.der*/
-    file_name[8] = (char)(slot_id+'0');
-
-    /*check the input parameter*/
-    if ((cert_chain == NULL) ^ (cert_chain_size == 0) ) {
-        return false;
-    }
-
-#if defined(_WIN32) || (defined(__clang__) && (defined (LIBSPDM_CPU_AARCH64) || \
-    defined(LIBSPDM_CPU_ARM)))
-    if ((fp_out = fopen(file_name, "w+b")) == NULL) {
-        printf("Unable to open file %s\n", file_name);
         return false;
-    }
-
-    if (cert_chain != NULL) {
-        if ((fwrite(cert_chain, 1, cert_chain_size, fp_out)) != cert_chain_size) {
-            printf("Write output file error %s\n", file_name);
-            fclose(fp_out);
+    } else {
+    #if defined(_WIN32) || (defined(__clang__) && (defined (LIBSPDM_CPU_AARCH64) || \
+        defined(LIBSPDM_CPU_ARM)))
+        FILE *fp_out;
+    #else
+        int64_t fp_out;
+    #endif
+
+        char file_name[] = "slot_id_0_cert_chain.der";
+        /*change the file name, for example: slot_id_1_cert_chain.der*/
+        file_name[8] = (char)(slot_id+'0');
+
+        /*check the input parameter*/
+        if ((cert_chain == NULL) ^ (cert_chain_size == 0) ) {
             return false;
         }
-    }
 
-    fclose(fp_out);
-#else
-    if (cert_chain != NULL) {
-        if ((fp_out = open(file_name, O_WRONLY | O_CREAT, S_IRWXU)) == -1) {
+    #if defined(_WIN32) || (defined(__clang__) && (defined (LIBSPDM_CPU_AARCH64) || \
+        defined(LIBSPDM_CPU_ARM)))
+        if ((fp_out = fopen(file_name, "w+b")) == NULL) {
             printf("Unable to open file %s\n", file_name);
             return false;
         }
 
-        if ((write(fp_out, cert_chain, cert_chain_size)) != cert_chain_size) {
-            printf("Write output file error %s\n", file_name);
-            close(fp_out);
-            return false;
+        if (cert_chain != NULL) {
+            if ((fwrite(cert_chain, 1, cert_chain_size, fp_out)) != cert_chain_size) {
+                printf("Write output file error %s\n", file_name);
+                fclose(fp_out);
+                return false;
+            }
         }
-    } else {
-        if ((fp_out = open(file_name, O_WRONLY | O_TRUNC)) == -1) {
-            printf("Unable to open file %s\n", file_name);
-            return false;
+
```

Comment:
Do we need to handle the (cert_chain == NULL) case?

Reply:
That's a separate issue. I just added
```
    if (g_set_cert_is_busy) {
        *is_busy = true;
```
and everything got adjusted. But yes, it looks like the "sample" implementation doesn't support erasing a certificate slot.

Reply:
Yes, the handling could be done in a separate change.